pkg/controller/controller.go

```diff
@@ -659,15 +654,9 @@ func (c *controller) getBrokerClientForServiceBinding(instance *v1beta1.ServiceI
 			return nil, err
 		}
 
-		var found bool
-		brokerClient, found = c.brokerClientManager.BrokerClient(NewClusterServiceBrokerKey(broker.Name))
-
-		if !found {
-			return nil, fmt.Errorf("OSB client not found for the broker %s", broker.Name)
-		}
+		brokerClient, err = c.clusterServiceBrokerClient(broker)
```

Comment:
check for non-nil err after this

Reply:
you are right, the check is missing, added